### PR TITLE
Strict mode

### PR DESCRIPTION
**Pull request by MonkeyBuddy**
Hi, I added some changes to raise exceptions instead of warning when no tenant is provided.

### CODE REVIEW

**Comment by s4ke** [comment]
Not the maintainer of this project, but I think the settings var should be prefixed with DJANGO_MULTITENANT or something along these lines as it will pollute the namespace of all apps that use it.

**Comment by MonkeyBuddy** [comment]
> Not the maintainer of this project, but I think the settings var should be prefixed with DJANGO_MULTITENANT or something along these lines as it will pollute the namespace of all apps that use it.

thanks for the suggestion, I added the prefix :)

**Review by JelteF — APPROVED** [review]
Thanks for the contribution! I left one small comment. Other than that this is good to merge I think.

File: django_multitenant/fields.py [hunk 1]
```diff
@@ -39,6 +41,8 @@ def get_extra_descriptor_filter(self, instance):
         current_tenant = get_current_tenant()
         if current_tenant:
             return get_tenant_filters(self.related_model)
+        elif getattr(settings, "TENANT_STRICT_MODE", False):
+            raise EmptyTenant
```
[the remark this answers is not in the document]
Comment (MonkeyBuddy) [reply django_multitenant/fields.py]: np, I changed the code, please have a look :)

File: django_multitenant/mixins.py [hunk 2]
```diff
@@ -80,6 +82,8 @@ def _do_update(self, base_qs, using, pk_val, values, update_fields, forced_updat
         if current_tenant:
             kwargs = get_tenant_filters(self.__class__)
             base_qs = base_qs.filter(**kwargs)
+        elif getattr(settings, "TENANT_STRICT_MODE", False):
+            raise EmptyTenant
```
Comment (JelteF) [inline django_multitenant/mixins.py:86]: Could you also set the message of the exception here?

**Review by JelteF — COMMENTED** [review]
If it's easy to add a test for this, could you add one?

**Review by JelteF — APPROVED** [review]

**Comment by MonkeyBuddy** [comment]
> If it's easy to add a test for this, could you add one?

sorry I was busy at work, I see you merged it. thanks.
I'll add another PR with another feature and tests for this one in the coming days :)

**Comment by JelteF** [comment]
No worries, thanks for the contribution. I'm looking forward to the PR with the tests.

**Pull request merged**